tools/osv-linter/internal/checks/checks.go

```diff
@@ -0,0 +1,43 @@
+package checks
+
+import "github.com/tidwall/gjson"
+
+type CheckCode string
+
+type CheckError struct {
+	Text     string
+}
+
+type Check struct {
+	Code        CheckCode
+	Name        string
+	Description string
+	Check       func(*gjson.Result) []CheckError
+}
+
+type CheckCollection struct {
+	Name        string
+	Description string
+	Checks      []*Check
+}
+
+var CheckIntroducedEventExists = &Check{
+	Code:        "R0001",
+	Name:        "introduced-event-exists",
+	Description: "every range has an introduced event",
+	Check:       RangeHasIntroducedEvent,
+}
+
+var AllChecks = map[string]*Check{
+	"introduced-event-exists": CheckIntroducedEventExists,
```

Comment:
This can probably be generated dynamically at runtime. Otherwise there's always going to be some duplication with the map key here and the check name itself.

Instead, we can just store a slice of Checks, from which the map is generated at runtime.

Reply:
Done in 750a53d